Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -24,33 +25,38 @@ where
 import qualified Control.Monad as CM
 import Control.Monad.Extra (maybeM)
 import qualified Data.HashMap as HM
+import Domain.Types.Configs.DriverPoolConfig
+import Domain.Types.Configs.TransporterConfig (TransporterConfig)
 import Domain.Types.Merchant (Merchant)
 import Domain.Types.Person (Driver)
 import qualified Domain.Types.SearchRequest as DSR
-import Domain.Types.TransporterConfig (TransporterConfig)
 import EulerHS.Prelude hiding (id)
 import Kernel.Randomizer (randomizeList)
 import Kernel.Storage.Esqueleto (EsqDBReplicaFlow)
 import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Error
 import Kernel.Types.Id
 import Kernel.Utils.Common
-import SharedLogic.Allocator.Jobs.SendSearchRequestToDrivers.Config (HasSendSearchRequestJobConfig)
 import SharedLogic.Allocator.Jobs.SendSearchRequestToDrivers.Handle.Internal.DriverPool.Config as Reexport
 import SharedLogic.DriverPool
-import qualified SharedLogic.DriverPool.Config as DP
-import Storage.CachedQueries.CacheConfig (HasCacheConfig)
-import qualified Storage.CachedQueries.TransporterConfig as TC
+import Storage.CachedQueries.CacheConfig (CacheFlow, HasCacheConfig)
+import qualified Storage.CachedQueries.Configs.DriverIntelligentPoolConfig as DIP
+import qualified Storage.CachedQueries.Configs.TransporterConfig as TC
+import qualified Storage.Queries.SearchRequest as QSR
 import Tools.Maps as Maps
 import Tools.Metrics
 
 isBatchNumExceedLimit ::
   ( Redis.HedisFlow m r,
-    HasSendSearchRequestJobConfig r
+    EsqDBFlow m r,
+    CacheFlow m r
   ) =>
   Id DSR.SearchRequest ->
   m Bool
 isBatchNumExceedLimit searchReqId = do
-  maxNumberOfBatches <- asks (.sendSearchRequestJobCfg.driverPoolBatchesCfg.maxNumberOfBatches)
+  searchReq <- QSR.findById searchReqId >>= fromMaybeM (SearchRequestNotFound searchReqId.getId) -- run these in replica
+  driverPoolConfig <- getDriverPoolConfig searchReq.providerId searchReq.estimatedDistance
+  let maxNumberOfBatches = driverPoolConfig.maxNumberOfBatches
```

Comment:
Take `maxNumberOfBatches` from top only. Instead of getting search_request and then getting driver pool configs.

Reply:
I think you should pass it from outside of this func like you do with `driverPoolConfig` and `getRescheduleTime` func for example